src/proto/proto_mesg_factory.cpp

```diff
@@ -30,6 +30,61 @@ namespace nuraft_mesg {
 std::string group_factory::m_ssl_cert;
 using handle_resp = std::function< void(RaftMessage&, ::grpc::Status&) >;
 
+static nuraft::cmd_result_code grpc_status_to_nuraft_code(::grpc::Status const& s) {
+    if (s.ok()) { return nuraft::cmd_result_code::OK; }
+    auto const ec = s.error_code();
+    switch (ec) {
+    case ::grpc::StatusCode::DEADLINE_EXCEEDED:
+        return nuraft::cmd_result_code::TIMEOUT;
+    case ::grpc::StatusCode::UNAVAILABLE:
+    case ::grpc::StatusCode::NOT_FOUND:
+        return nuraft::cmd_result_code::SERVER_NOT_FOUND;
+    case ::grpc::StatusCode::CANCELLED:
+    case ::grpc::StatusCode::ABORTED:
+        return nuraft::cmd_result_code::CANCELLED;
+    case ::grpc::StatusCode::FAILED_PRECONDITION:
+        return nuraft::cmd_result_code::TERM_MISMATCH;
+    case ::grpc::StatusCode::ALREADY_EXISTS:
+        return nuraft::cmd_result_code::SERVER_ALREADY_EXISTS;
+    case ::grpc::StatusCode::INVALID_ARGUMENT:
+    case ::grpc::StatusCode::UNIMPLEMENTED:
+    case ::grpc::StatusCode::UNAUTHENTICATED:
+    case ::grpc::StatusCode::PERMISSION_DENIED:
+    case ::grpc::StatusCode::RESOURCE_EXHAUSTED:
+    case ::grpc::StatusCode::OUT_OF_RANGE:
+        return nuraft::cmd_result_code::BAD_REQUEST;
+    case ::grpc::StatusCode::DATA_LOSS:
+    default:
+        return nuraft::cmd_result_code::FAILED;
+    }
+}
+
+static constexpr bool is_powerof2(uint64_t v) { return v && ((v & (v - 1)) == 0); }
+
+static void log_every_nth(std::string const& addr, ::grpc::Status const& status, std::string const& msg_type) {
+    static thread_local std::unordered_map< std::string, std::pair< uint64_t, Clock::time_point > > t_errors;
+    static constexpr uint64_t every_nth_sec = 60;
+    std::string msg = addr + "-" + status.error_message();
+
+    uint64_t failed_count = 0;
```

Comment:
`= 1` to remove the two other assignments below?

Reply:
updated that, please check